mahiru/components/network_administrator.py

```diff
@@ -46,7 +71,21 @@ def serve_asset(
         Return:
             A connection info object to send back to the client.
         """
-        raise NotImplementedError()
+        if not isinstance(request, WireGuardConnectionRequest):
```

Comment:
We'd get in trouble with professor Liskov. She says that since the interface declares that we'll accept a `ConnectionRequest`, an implementing class that only accepts a `WireGuardConnectionRequest` doesn't fully implement the interface. Mypy actually gives a very nice warning:
```
mahiru/components/network_administrator.py:59: error: Argument 3 of "serve_asset" is incompatible with supertype "INetworkAdministrator"; supertype defines the argument type as "ConnectionRequest"
mahiru/components/network_administrator.py:59: note: This violates the Liskov substitution principle
mahiru/components/network_administrator.py:59: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

However, I can make the return type more specific (https://en.wikipedia.org/wiki/Covariant_return_type), I'll do that.